### ISSUE TITLE
ddp no_sync

### ISSUE THREAD

**Issue by ruotianluo** [PyTorchLightning/pytorch-lightning#2595]
## 🚀 Feature
DDP has this no_sync context manager.
https://pytorch.org/docs/stable/nn.html?highlight=no_sync#torch.nn.parallel.DistributedDataParallel.no_sync

Could be good to add it when doing gradient accumulation.
